site/src/components/Dashboard/DashboardProvider.tsx

```diff
@@ -0,0 +1,89 @@
+import { useMachine } from "@xstate/react"
+import {
+  AppearanceConfig,
+  BuildInfoResponse,
+  Entitlements,
+  Experiments,
+} from "api/typesGenerated"
+import { FullScreenLoader } from "components/Loader/FullScreenLoader"
+import { createContext, FC, PropsWithChildren, useContext } from "react"
+import { appearanceMachine } from "xServices/appearance/appearanceXService"
+import { buildInfoMachine } from "xServices/buildInfo/buildInfoXService"
+import { entitlementsMachine } from "xServices/entitlements/entitlementsXService"
+import { experimentsMachine } from "xServices/experiments/experimentsMachine"
+
+interface Appearance {
+  config: AppearanceConfig
+  preview: boolean
+  setPreview: (config: AppearanceConfig) => void
+  save: (config: AppearanceConfig) => void
+}
+
+interface DashboardProviderValue {
+  buildInfo: BuildInfoResponse
+  entitlements: Entitlements
+  appearance: Appearance
+  experiments: Experiments
+}
+
+export const DashboardProviderContext = createContext<
+  DashboardProviderValue | undefined
+>(undefined)
+
+export const DashboardProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [buildInfoState] = useMachine(buildInfoMachine)
+  const [entitlementsState] = useMachine(entitlementsMachine)
+  const [appearanceState, appearanceSend] = useMachine(appearanceMachine)
+  const [experimentsState] = useMachine(experimentsMachine)
+  const { buildInfo } = buildInfoState.context
+  const { entitlements } = entitlementsState.context
+  const { appearance, preview } = appearanceState.context
+  const { experiments } = experimentsState.context
+  const isLoading = !buildInfo || !entitlements || !appearance || !experiments
```

Comment:
What happens if one of these is undefined because of an error?

Reply:
If they are undefined, we are going to show the loading state. About the error, good catch, I forgot to handle it.

Reply:
Not sure if this can apply here, but I fixed a bug recently where the value had been fetched but was falsy so a check like this left us in the loading state forever. Just something to keep in mind, whether falsiness really indicates loading. I like to put "loading" tags on loading states in the machines.

Reply:
Interesting case, but in theory, any of these data should be undefined or even false.

Reply:
Buuuut, this should not be here. This is going to be refactored for something like this: https://codercom.slack.com/archives/CJURPL8DN/p1674072426535449?thread_ts=1674071396.499819&cid=CJURPL8DN